streams-contrib/streams-persist-elasticsearch/src/main/java/org/apache/streams/elasticsearch/ElasticsearchPersistReader.java

```diff
@@ -49,35 +26,73 @@
-    private ElasticsearchReaderConfiguration config;
-    private int threadPoolSize = 10;
-    private ExecutorService executor;
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
-    private Future<?> readerTask;
+    private final ElasticsearchReaderConfiguration config;
+    private final ElasticsearchClientManager elasticsearchClientManager;
+    private final AtomicBoolean isRunning = new AtomicBoolean(false);
 
-    public ElasticsearchPersistReader() {
+    public ElasticsearchPersistReader(ElasticsearchReaderConfiguration config) {
+        this(config, new ElasticsearchClientManager(config));
     }
 
-    public ElasticsearchPersistReader(ElasticsearchReaderConfiguration config) {
+    public ElasticsearchPersistReader(ElasticsearchReaderConfiguration config, ElasticsearchClientManager escm) {
         this.config = config;
+        this.elasticsearchClientManager = escm;
+    }
+
+    @Override
+    public boolean isRunning() {
+        return this.isRunning.get();
     }
 
-    //PersistReader methods
     @Override
     public void startStream() {
         LOGGER.debug("startStream");
-        executor = Executors.newSingleThreadExecutor();
-        readerTask = executor.submit(new ElasticsearchPersistReaderTask(this, elasticsearchQuery));
+        this.isRunning.set(true);
+        final ElasticsearchQuery query = this.elasticsearchQuery;
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    while (!query.isCompleted()) {
+                        if(query.hasNext()) {
+                            SearchHit hit = query.next();
+                            StreamsDatum item = new StreamsDatum(hit.getSourceAsString(), hit.getId());
+                            item.getMetadata().put("id", hit.getId());
+                            item.getMetadata().put("index", hit.getIndex());
+                            item.getMetadata().put("type", hit.getType());
+                            ComponentUtils.offerUntilSuccess(item, streamsResultSet.getQueue());
+                        }
+                        else {
+                            try {
+                                Thread.sleep(1);
+                            }
+                            catch(InterruptedException ioe) {
+                                LOGGER.error("sleep error: {}", ioe);
+                            }
+                        }
+                    }
+                }
+                catch(Throwable e) {
+                    LOGGER.error("Unexpected issue: {}", e.getMessage());
+                }
+                finally{
+                    isRunning.set(false);
+                }
+            }
+        }).start();
     }
 
     @Override
     public void prepare(Object o) {
-        elasticsearchQuery = this.config == null ? new ElasticsearchQuery() : new ElasticsearchQuery(config);
+        if(this.config == null)
```

Comment:
Yes, it's a good convention to use.  Curly braces on all conditionals makes the code more maintainable and only costs two keystrokes. :)